### PR TITLE
[AUT-5188] single command to setup graphql demo

### PR DESCRIPTION
**Pull request by morus12**
## Description

Added `make graphql-demo` that sets up GraphQL demo with:
- installed ACP
- installed Istio and Istio Authorizer
- installed Countries GraphQL service

Screenshot 2022-02-02 at 09-28-57 Playground - http countries ingress k8s 9080

## Type of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Tests (extending the test suite)
- [ ] Refactor (internal improvement that doesn't change product functionality)
- [x] Other (if none of the other choices apply)

### CODE REVIEW

File: examples/graphql-demo/parse-body.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+apiVersion: networking.istio.io/v1alpha3
```
Comment (holowinski) [inline examples/graphql-demo/parse-body.yaml:1]: There's a new extension to `istio-authorizer` helm chart which does this part.

Have a look at https://github.com/cloudentity/acp-helm-charts/pull/137
It probably makes sense to use this when the PR gets merged.

**Review by holowinski — CHANGES_REQUESTED** [review]

File: examples/graphql-demo/istio-configmap.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+y=yaml.safe_load(sys.stdin.read())
+
+y['data']['mesh'] += """
+extensionProviders:
+- name: "acp-authorizer"
+  envoyExtAuthzGrpc:
+    service: "istio-authorizer.acp-system.svc.cluster.local"
+    port: "9001"\n"""
+
+yaml.safe_dump(y, sys.stdout)
```
Comment (ikawalec) [inline examples/graphql-demo/istio-configmap.py:19]: Potentially, you could use istio operator: https://github.com/cloudentity/acp-saas/blob/master/apps/base/istio-tests/istio_operator.yaml#L28 to install istio and apply configmap changes.

File: values/istio-authorizer.yaml [hunk 3]
```diff
@@ -2,3 +2,5 @@ clientCredentials:
   clientID: "my_test_gateway_client"
   clientSecret: "5c8f3f5b353ba0f55c8f3f5b353ba0f5"
 issuerURL: "https://acp.acp-system:8443/system/system"
+extraEnvVars:
+  DO_NOT_FAIL_ON_NON_MATCHING_REQUESTS: "true"
```
Comment (ikawalec) [inline values/istio-authorizer.yaml:6]: why this is required?

**Review by ikawalec — CHANGES_REQUESTED** [review]

File: values/istio-authorizer.yaml (hunk 3 above)
Comment (morus12) [reply values/istio-authorizer.yaml:6]: Countries demo exposes UI for querying GraphQL API. 
Istio Authorizer doesn't support mixed mode (GraphQL + REST) for a single service.

File: values/istio-authorizer-graphql.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  clientID: "my_test_gateway_client"
+  clientSecret: "5c8f3f5b353ba0f55c8f3f5b353ba0f5"
+issuerURL: "https://acp.acp-system:8443/system/system"
+parseBody:
+  enabled: true
+data:
+  config: |
+    http_client:
+      root_ca: /data/ca.crt
+    do_not_fail_on_non_matching_requests: true
```
Comment (ikawalec) [inline values/istio-authorizer-graphql.yaml:11]: FYI with 2.x release, this will be changed to:

```
enforcement:
  allowUnknown: true
```

**Review by ikawalec — APPROVED** [review]

**Review by holowinski — APPROVED** [review]

**Pull request merged**